### PR TITLE
IEN - 537 continued 

### PR DESCRIPTION
**Pull request by dbayly-freshworks**
Added expiry to tokens and added date filter to records

### CODE REVIEW

File: apps/api/src/applicant/external-api.service.ts [hunk 1]
```diff
@@ -628,16 +631,26 @@ export class ExternalAPIService {
...
-        'applicant_audit',
-      ],
-    });
+
+  
+  async getApplicants(start?:string,end?:string ) {
+      return this.ienapplicantRepository.find({
+        where: {
+          updated_date: Between(
+            new Date(start ||'1918-07-18' ).toISOString(),
```
Comment (sunghwan-park-fw) [inline apps/api/src/applicant/external-api.service.ts:640]: Does this date have special meaning?

Comment (dbayly-freshworks) [reply apps/api/src/applicant/external-api.service.ts:640]: It's the day World war 1 was declared. I needed an arbitrary date that should be well before any record is updated. 
It's an easter egg, and I can remove it if you like.

File: apps/api/src/auth/jwt.guard.ts [hunk 2]
```diff
@@ -41,11 +42,20 @@ export class JWTGuard implements CanActivate {
...
         process.env.JWT_SECRET ? process.env.JWT_SECRET : 'jwtsecret',
-      );
-      return decoded;
+      )) as any;
+      if(!decoded.exp){
+        return false; 
+      }
+      // Check token expiry
+      // JWT Token expiry is in seconds since the unix epoc, so we need to multiply them by 1000 to convert them into timestamps.
+      if(dayjs(decoded?.exp*1000).isAfter(dayjs()) && dayjs(decoded?.exp*1000).isBefore(dayjs().add(5,'minutes')) ){
```
Comment (sunghwan-park-fw) [inline apps/api/src/auth/jwt.guard.ts:54]: Is this to enforce token to expire in 5 minutes?

Comment (dbayly-freshworks) [reply apps/api/src/auth/jwt.guard.ts:54]: That is correct. I check if the expiry is after the current time, but before five minutes in the future.

**Review by sunghwan-park-fw — APPROVED** [review]

**Pull request merged**